
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.3" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">
<link rel="stylesheet" type="text/css" href="../css/readable.css">
<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs-1.1/textmate.css"
      type="text/css" />

<script src="../site_libs/highlightjs-1.1/highlight.js"></script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/toc2.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_HTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Divvy data exploration project</title>

<style type = "text/css">
body {
  
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Divvy data exploration project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../setup.html">Setup</a>
</li>
        
<li>
  <a href="../license.html">License</a>
</li>
        
<li>
  <a href="../notes.html">Notes</a>
</li>
        
<li>
  <a href="../protected.html">Protected</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/stephenslab/ipynb-website"> source </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Stationarity-distribution-of-the-discrete-time-stepping-stone-model">Stationarity distribution of the discrete time stepping stone model<a class="anchor-link" href="#Stationarity-distribution-of-the-discrete-time-stepping-stone-model">&#182;</a></h1><p><strong>authors:</strong> Joseph Marcus</p>
<p>Here I explore the stepping stone model with possible approximations building off the classic results of Bodmer and Cavalli-Sforza 1967. I essentially find there is no clean analytical form for the covariance matrix of the stationary distribution of this process.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Discrete-time-stepping-stone">Discrete time stepping stone<a class="anchor-link" href="#Discrete-time-stepping-stone">&#182;</a></h2><p>Consider a single bi-allelic SNP with haploid individuals carrying either the $A$ or $a$ allele dispersed throughout a habitat. The habitat is discretized and defined on a graph $\mathcal{G}$ over geographic space with $d$ nodes and a migration matrix $\mathbf{M}$ which specifies the edge weights. Note that $\mathbf{M}$ can be interpreted as a "backwards" migration matrix where $m_{ij} &gt;= 0.0$ and $\sum_{j=1}^d m_{ij} = 1$. Furthermore, $m_{ij}$ can be interpreted as the probability that an individual in node $i$ has parents from node $j$. Let $p_{i,t}$ be the allele frequency of the $A$ allele at node $i$ and time $t$, here time is discrete as well. Each generation we can describe the evolution of the allele frequency in two steps, first a deterministic migration event where individuals are swapped amongst only neighboring nodes and a drift event which is a random fluctuation in allele frequency in each node proportional to its population size.</p>
<p>$$
p_{i,t} = \sum_{j=1}^d m_{ij} p_{i,t-1}
$$</p>
<p>Or in matrix notation</p>
<p>$$
\mathbf{p}_t = \mathbf{M}\mathbf{p}_{t-1}
$$</p>
<p>For now we don't assume any distributional form for $\mathbf{p}_{t}$ but do define its conditional moments</p>
<p>$$
\begin{aligned}
E\big(\mathbf{p}_t | \mathbf{p}_{t-1}\big) &amp;= \mathbf{M}\mathbf{p}_{t-1} \\
Var\big(\mathbf{p}_t | \mathbf{p}_{t-1}\big) &amp;= diag\Big(\frac{1}{\mathbf{N}} \odot \mathbf{M}\mathbf{p}_{t-1} \odot \big(\mathbf{1} - \mathbf{M}\mathbf{p}_{t-1}\big) \Big)
\end{aligned}
$$</p>
<p>Here $\mathbf{N}$ is a $d$ vector of population sizes within each node and $\odot$ refers to element-wise multiplication. Note that this exactly corresponds to the process we described previously. There is first a deterministic migration event and variance induced by random sampling of gametes due to genetic drift. Here we make the simplifying assumption where we focus only on common SNPs such that the binomial sampling variance has a small range and approximate this conditional variance as</p>
<p>$$
Var\big(\mathbf{p}_t | \mathbf{p}_{t-1}\big) \approx \sigma^2 diag\Big(\frac{1}{\mathbf{N}}\Big)
$$</p>
<p>Now lets make a further assumption that the change in frequency due to drift are normally distributed</p>
<p>$$
\mathbf{p}_t = \mathbf{M}\mathbf{p}_{t-1} + \epsilon \\ 
\epsilon | \sigma^2, \mathbf{N} \sim \mathcal{N}\Bigg(\mathbf{0}, \sigma^2 diag\Big(\frac{1}{\mathbf{N}}\Big)\Bigg) 
$$</p>
<p>For notational simplicity let $\mathbf{Q} = \sigma^2diag\Big(\frac{1}{\mathbf{N}}\Big)$</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Stationary-distribution">Stationary distribution<a class="anchor-link" href="#Stationary-distribution">&#182;</a></h2><p>Let $\pi(.)$ be the stationary distribution of this process and $\mathbf{p}^{(s)}$ be a draw from the stationary distribution $\mathbf{p}^{(s)} \sim \pi(.)$. It might be reasonable to assume that $\mathbf{p}^{(s)} \sim \mathcal{N}\big(\mu, \mathbf{\Sigma}\big)$. Then the distribution of $\mathbf{M}\mathbf{p}^{(s)} + \epsilon$ will have covariance matrix $\mathbf{\Sigma}$ because we are at stationarity, resulting in ...</p>
<p>$$
\mathbf{\Sigma} = \mathbf{M}\mathbf{\Sigma}\mathbf{M}^T + \mathbf{Q}
$$</p>
<p>Interestingly, this is known as the discrete time <a href="https://en.wikipedia.org/wiki/Lyapunov_equation">Lyapunov equation</a> and has a limiting solution (under certain conditions for $\mathbf{M}$)</p>
<p>$$
\mathbf{\Sigma} = \sum_{k=0}^{\infty} \mathbf{M}^K\mathbf{Q}(\mathbf{M}^k)^T
$$</p>
<p>the Lyapunov equation has a more intuitive interpretation if we take a different approach in finding the stationary covariance $\Rightarrow$</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Limiting-distribution">Limiting distribution<a class="anchor-link" href="#Limiting-distribution">&#182;</a></h2><p>Recall that</p>
<p>$$
Var\big(\mathbf{p}_t | \mathbf{p}_{t-1}\big) \approx \mathbf{Q}
$$</p>
<p>Lets find the marginal covariance of the allele frequency at time $t$ by the law of total variance</p>
<p>$$
\begin{aligned}
Var(\mathbf{p}_t) &amp;= E\Big(Var\big(\mathbf{p}_t | \mathbf{p}_{t-1}\big)\Big) +  Var\Big(E\big(\mathbf{p}_{t} | \mathbf{p}_{t-1} \big)\Big) \\
&amp;= E\big(\mathbf{Q}\big) + Var\big(\mathbf{M}\mathbf{p}_{t-1}\big) \\
&amp;= \mathbf{Q} + \mathbf{M}Var(\mathbf{p}_{t-1})\mathbf{M}^T
\end{aligned}
$$</p>
<p>$$
\begin{aligned}
\dots \\
&amp;= \mathbf{Q} + \mathbf{M}Var(\mathbf{p}_{t-1})\mathbf{M}^T \\
&amp;= \mathbf{Q} + \mathbf{M}\Big(\mathbf{Q} + \mathbf{M}Var(\mathbf{p}_{t-2})\mathbf{M}^T\Big)\mathbf{M}^T \\
&amp;= \mathbf{Q} + \mathbf{M}\mathbf{Q}\mathbf{M}^T + \mathbf{M}^2 Var(\mathbf{p}_{t-2})(\mathbf{M}^2)^T \\
&amp;= \mathbf{Q} + \mathbf{M}\mathbf{Q}\mathbf{M}^T + \mathbf{M}^2\Big(\mathbf{Q} + \mathbf{M}Var(\mathbf{p}_{t-3})\mathbf{M}^T\Big)(\mathbf{M}^2)^T \\
&amp;= \dots \\
&amp;= \mathbf{Q} + \mathbf{M}\mathbf{Q}\mathbf{M}^T  + \mathbf{M}^2\mathbf{Q}(\mathbf{M}^2)^T + \mathbf{M}^3\mathbf{Q}(\mathbf{M}^3)^T + \dots + \mathbf{M}^t\mathbf{Q}\mathbf{M}^t)^T \\
&amp;= \sum_{k=0}^t \mathbf{M}^k\mathbf{Q}(\mathbf{M}^k)^T
\end{aligned}
$$</p>
<p>If now let the process evolve for infinite time and we recognize the solution to the Lyapunov equation</p>
<p>$$
Var\big(\mathbf{p}^{(s)}\big) = \sum_{k=0}^{\infty} \mathbf{M}^K\mathbf{Q}(\mathbf{M}^k)^T
$$</p>
<p>The convergence properties of this infinite sum requires that $|\lambda_i| &lt; 1 \ \forall \ i$ where $\lambda_i$ is an eigen-value of $\mathbf{M}$. Me know that $\mathbf{M}$ is stochastic matrix and as such has an eigen-value that is exactly 1 thus there doesn't seem a tractable analytical form for the stationarity covariance of this process.</p>

</div>
</div>
</div>
<hr>
&copy; 2017 Peter Carbonetto &amp; Gao Wang
<p><small>Exported from <a href="http://github.com/stephenslab/ipynb-website/blob/cb98621313ea5f296198677fb77339daaead55c6/analysis/stepping-stone-stationary.ipynb"><code>analysis/stepping-stone-stationary.ipynb</code></a> committed by Gao Wang on Mon Feb 12 10:22:58 2018 <a href="http://github.com/stephenslab/ipynb-website/commit/cb98621313ea5f296198677fb77339daaead55c6">revision 1, cb98621</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
